tests/trainer/properties/test_auto_gpu_select.py

```diff
@@ -36,10 +35,11 @@ def test_trainer_with_gpus_options_combination_at_available_gpus_env(auto_select
                 "auto_select_gpus=True, gpus=0 is not a valid configuration."
                 " Please select a valid number of GPU resources when using auto_select_gpus."
             ),
-        ):
+        ), pytest.deprecated_call(match=r"is deprecated in v1.6 and will be removed in v2.0."):
             Trainer(auto_select_gpus=auto_select_gpus, gpus=gpus)
     else:
-        Trainer(auto_select_gpus=auto_select_gpus, gpus=gpus)
+        with pytest.deprecated_call(match=r"is deprecated in v1.6 and will be removed in v2.0."):
```

Comment:
I guess we would need both versions or make a small modification to the above error message

> auto_select_gpus=True, gpus=0 is not a valid configuration.

so that it says devices.

Reply:
When I do this I am getting an error:
```
pytorch_lightning/utilities/argparse.py:339: in insert_env_defaults
    return fn(self, **kwargs)
pytorch_lightning/trainer/trainer.py:486: in __init__
    self._accelerator_connector = AcceleratorConnector(
pytorch_lightning/trainer/connectors/accelerator_connector.py:196: in __init__
    self._set_parallel_devices_and_init_accelerator()
pytorch_lightning/trainer/connectors/accelerator_connector.py:514: in _set_parallel_devices_and_init_accelerator
    self._parallel_devices = self.accelerator.get_parallel_devices(self._devices_flag)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

devices = None

    @staticmethod
    def get_parallel_devices(devices: List[int]) -> List[torch.device]:
        """Gets parallel devices for the Accelerator."""
        print("DEVICES")
        print(devices)
>       return [torch.device("cuda", i) for i in devices]
E       TypeError: 'NoneType' object is not iterable

pytorch_lightning/accelerators/gpu.py:84: TypeError
```
so this will require more investigation.